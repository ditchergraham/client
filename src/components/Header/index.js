import { h, Component } from 'preact';
import PropTypes from 'prop-types';

import { applyTheme } from '../ThemeProvider/';

import './style.css';

/**
* Header
*
*/
class Header extends Component {
  propTypes = {
    toggleChat: PropTypes.func,
    style: PropTypes.string,
    theme: PropTypes.shape({
      primary_colour: PropTypes.string.isRequired,
    }),
    chatOpen: PropTypes.boolean,
  };

  renderToggleChatButton = () => (this.props.chatOpen ? <span>Ã—</span> : <span>...</span>);

  render () {
    const { toggleChat, style, theme } = this.props;

    return (
      <header
        style={{ backgroundColor: theme.primary_colour }}
        className={`Header--${style}`}
        onClick={() => toggleChat(true)}
      >
        <span className={`Header__title--${style}`}>Chat with John</span>
        <button className={`Header__closeBtn--${style}`} onClick={() => toggleChat(true)}>
          {this.renderToggleChatButton()}
        </button>
      </header>
    );
  }
}

export default applyTheme(Header);
