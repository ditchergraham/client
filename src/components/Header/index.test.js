import { h } from 'preact';
import render from 'preact-render-to-string';

import Header from '.';
import { ThemeProvider } from '../ThemeProvider';

describe('<Header />', () => {
  it('should render ', () => {
    const theme = {
      primary_colour: 'test',
    };
    const tree = render(<ThemeProvider theme={theme} style="messenger">
      <Header theme={theme} />
    </ThemeProvider>);
    expect(tree).toMatchSnapshot();
  });
});
