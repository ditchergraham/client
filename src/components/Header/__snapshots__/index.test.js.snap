// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render  1`] = `"<header style=\\"background-color: test;\\" class=\\"Header--messenger\\"><span class=\\"Header__title--messenger\\">Chat with John</span><button class=\\"Header__closeBtn--messenger\\"><span>...</span></button></header>"`;
